healthcheck/client.go

```diff
@@ -3,29 +3,25 @@ package healthcheck
 import (
 	"errors"
 	"sync"
-
-	rchttp "github.com/ONSdigital/dp-rchttp"
 )
 
-// Client represents a healthcheck client
+// Client represents a health check client
 type Client struct {
-	Clienter rchttp.Clienter
-	Check    *Check
-	Checker  *Checker
-	mutex    *sync.Mutex
+	Check   *Check
+	Checker *Checker
+	mutex   *sync.Mutex
 }
 
-// NewClient returns a pointer to a new instantiated Client with
+// newClient returns a pointer to a new instantiated Client with
 // the provided checker function and an rchttp.Clienter
-func NewClient(cli rchttp.Clienter, checker *Checker) (*Client, error) {
-	if cli == nil {
-		return nil, errors.New("expected clienter but none provided")
+func newClient(checker *Checker) (*Client, error) {
+	if checker == nil {
+		return nil, errors.New("expected checker but none provided")
 	}
 
 	return &Client{
-		Clienter: cli,
-		Check:    nil,
-		Checker:  checker,
-		mutex:    &sync.Mutex{},
+		Check:   nil,
```

Comment:
Just realised this is why we need to return the check object from checker. As the health check client is unable to access the service/app/mongo etc. client which will always contain a Check.